units/src/amount/mod.rs

```diff
@@ -295,12 +297,19 @@ fn parse_signed_to_satoshi(
     // Decimally shift left by `max_decimals - decimals`.
     let scale_factor = max_decimals - decimals.unwrap_or(0);
     for _ in 0..scale_factor {
-        value = match 10_u64.checked_mul(value) {
+        value = match 10_i64.checked_mul(value) {
             Some(v) => v,
             None => return Err(InnerParseError::Overflow { is_negative }),
         };
     }
 
+    if is_negative {
+        value = match value.checked_mul(-1) {
+            Some(v) => v,
+            None => return Err(InnerParseError::Overflow { is_negative }),
+        }
+    }
```

Comment:
How about, instead of `checked_mul` we try to construct `SignedAmount`, then negate it if it's negative and return it? I think it better conveys the intention and removes a branch which might be difficult to optimize-out.

Reply:
I'm unsure. If we directly try to create a `SignedAmount` and then negate it, then we're checking whether the negative of the value exceeds `SignedAmount::MAX`. Whereas conceptually we should be checking whether the value is lower than `SignedAmount::MIN`.

In practice these checks are identical because `MIN` is the negative of `MAX`. But I made a small effort to make sure the code was still correct even if that changed. (Although, given that we have an infallible negation operator, maybe the situation is forced.)

Reply:
I think `MIN` will never change *also* for the reason of having infallible negation. (And if we did change the negation to become fallible it'd show at the place where we'd be calling it)

But I just realized we also need `map_err` to not return funny error type claiming we tried to parse negative. Still, it seems clearer code to me.

Reply:
Changed to use your suggestion. This does simplify the `from_str_in` call sites a fair bit.